src/Esfa.Vacancy.Register.Infrastructure/Services/TrainingDetailService.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Threading.Tasks;
+using Esfa.Vacancy.Register.Domain.Entities;
+using Esfa.Vacancy.Register.Domain.Interfaces;
+using SFA.DAS.Apprenticeships.Api.Client;
+
+namespace Esfa.Vacancy.Register.Infrastructure.Services
+{
+    public class TrainingDetailService : ITrainingDetailService
+    {
+        public async Task<Framework> GetFrameworkDetails(int code)
+        {
+            using (var client = new FrameworkCodeClient())
+            {
+                var framework = await client.GetAsync(code);
+                if (framework == null) throw new TrainingResourceNotFoundException($"Framework: {code}");
+                return new Framework() { Title = framework.Title };
+            }
+        }
+
+        public async Task<Standard> GetStandardDetails(int code)
+        {
+            using (var client = new StandardApiClient())
+            {
+                var standard = await client.GetAsync(code);
+                if (standard == null) throw new TrainingResourceNotFoundException($"Standard: {code}");
```

Comment:
One can say that this is hypothetical, however since we are querying an external api, there is no guarantee as yet that our systems are in sync. I could just use a system exception here instead of custom one, in either case the user is going to see 500 response.